src/lammpsio/box.py

```diff
@@ -63,6 +63,55 @@ def cast(cls, value):
         else:
             raise TypeError(f"Unable to cast boxlike object with shape {v.shape}")
 
+    @classmethod
+    def from_matrix(cls, low, matrix):
+        """Create a Box from low and matrix.
+
+        Parameters
+        ----------
+        low : list
+            Origin of the box.
+        matrix : :class:`numpy.ndarray`
+            Upper triangular matrix in LAMMPS style:
+            [[lx, xy, xz],
+             [0, ly, yz],
+             [0, 0, lz]]
+
+        Returns
+        -------
+        :class:`Box`
+            A simulation box.
+
+        Raises
+        ------
+        TypeError
+            If `low` is not length 3.
+        TypeError
+            If `matrix` is not a 3x3 array.
+        ValueError
+            If `matrix` is not upper triangular.
+
+        """
+        low = numpy.array(low, dtype=float)
+        arr = numpy.array(matrix, dtype=float)
+
+        if low.shape != (3,):
+            raise TypeError("Low must be a 3-tuple")
+        if arr.shape != (3, 3):
+            raise TypeError("Box matrix must be a 3x3 array")
+        if arr[1, 0] != 0 or arr[2, 0] != 0 or arr[2, 1] != 0:
+            raise ValueError("Box matrix must be upper triangular")
+
+        # Extract diagonal elements for box lengths
+        lx, ly, lz = arr[0, 0], arr[1, 1], arr[2, 2]
+        high = low + [lx, ly, lz]
+
+        # Extract tilt factors
+        xy, xz, yz = arr[0, 1], arr[0, 2], arr[1, 2]
+        tilt = [xy, xz, yz] if numpy.any([xy, xz, yz]) else None
```

Comment:
This is one area of LAMMPS that is slightly grey. You *may* want to have a triclinic box that has zero tilt factors, say in order to allow the box to deform later.

Should we add a default argument like `check_orthorhombic=True` or `force_triclinic=False` that will cause this check to occur? And otherwise, accept the tilts as they come from the matrix?